salt/modules/kubernetesmod.py

```diff
@@ -1593,3 +1593,13 @@ def __enforce_only_strings_dict(dictionary):
         ret[six.text_type(key)] = six.text_type(value)
 
     return ret
+
+
+def __base64_str(string):
+    '''
+    Converts a string of data into a base64 encoded string based on python version
+    '''
+    if sys.version_info >= (3, 0):
+        return str(base64.b64encode(string.encode('utf-8')), 'utf-8')
```

Comment:
How do we know this will always be `utf-8`?

Reply:
@dwoz If `string` in this case is already a `str` object and I'm simply trying to get it to a bytes representation so that can be passed to the `b64encode` call, does it matter what the underlying encoding of the string is? Perhaps I am misunderstanding the problem. Are you suggesting there may be a problem if there are characters in `string` that cannot be encoded by `utf-8`?

Reply:
@morinap There are 2 more salty ways to do this depending on your goals. If you need a `unicode` string as a result on both py2 and py3 you can use
```python
salt.utils.hashutils.base64_b64encode(string)
```
If you need a native `str` object on all platforms you can do either
```python
salt.utils.stringutils.to_str(salt.utils.hashutils.base64_b64encode(string))
```